x/incentive/keeper/keeper_lps.go

```diff
@@ -0,0 +1,58 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ctypes "github.com/elys-network/elys/x/commitment/types"
+)
+
+// Calculate new Eden token amounts based on LpElys committed and MElys committed
+func (k Keeper) CalculateRewardsForLPs(ctx sdk.Context, totalProxyTVL sdk.Dec, commitments ctypes.Commitments, edenAmountPerEpochLp sdk.Int, dexRevenueAmtForLPs sdk.Dec) (sdk.Int, sdk.Int) {
+	// Method 2 - Using Proxy TVL
+	totalNewEdenAllocated := sdk.ZeroInt()
+	totalDexRewardsAllocated := sdk.ZeroDec()
+
+	// Iterate to calculate total Eden from LpElys, MElys committed
+	k.lpk.IterateLiquidityPools(ctx, func(l LiquidityPool) bool {
+		// ------------ New Eden calculation -------------------
+		// -----------------------------------------------------
+		// newEdenAllocated = 80 / ( 80 + 90 + 200 + 0) * 100
+		// Pool share = 80
+		// edenAmountPerEpochLp = 100
+
+		// Calculate Proxy TVL share considering multiplier
+		proxyTVL := sdk.NewDecFromInt(l.TVL).MulInt64(l.multiplier)
+		poolShare := proxyTVL.Quo(totalProxyTVL)
+
+		// Calculate new Eden for this pool
+		newEdenAllocatedForPool := poolShare.MulInt(edenAmountPerEpochLp)
+
+		// this lp token committed
+		commmittedLpToken := commitments.GetCommittedAmountForDenom(l.lpToken)
+		// this lp token total committed
+		totalCommittedLpToken := k.tci.TotalLpTokensCommitted[l.lpToken]
+
+		// Calculalte lp token share of the pool
+		lpShare := sdk.NewDecFromInt(commmittedLpToken).QuoInt(totalCommittedLpToken)
+
+		// Calculate new Eden allocated per LP
+		newEdenAllocated := lpShare.Mul(newEdenAllocatedForPool).TruncateInt()
+
+		// Sum the total amount
+		totalNewEdenAllocated = totalNewEdenAllocated.Add(newEdenAllocated)
+		// -------------------------------------------------------
+
+		// ------------------- DEX rewards calculation -------------------
+		// ---------------------------------------------------------------
+		// Calculate dex rewards per pool
+		dexRewardsAllocatedForPool := poolShare.Mul(dexRevenueAmtForLPs)
```

Comment:
DEX reward for this pool will need to be updated to use `BalanceOf(RewardPool)` address.

Reply:
Or it's totally fine to keep as it is I guess.

Reply:
Okay got it. thanks.